### DISCUSSION TITLE
How to keep `openai>=1,<1.47` while upgrading `numpy` in tact

### DISCUSSION THREAD

**Discussion by jamesbraza** [renovatebot/renovate#31646] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

I have a Python package pinned like so:

```txt
openai>=1,<1.47  # Allow all of v1 before 1.47
tensorboard==2.16.2
```

With `rangeStrategy: "bump"`, Renovate does this, which to me is undesirable because it both exceeds 1.47 and stops allowing most of `openai` v1:

```txt
openai>=1.49.0,<1.50
tensorboard==2.18.0
```

With `rangeStrategy: "widen"` or `"replace"`, Renovate does this, which to me is undesirable because the `openai` upper bound exceeds 1.47:

```txt
openai>=1,<1.50
tensorboard==2.18.0
```

I like that it upgraded the equality pinned `tensorboard` in both cases.

I should say I am trying to avoid making special cases (e.g. with `allowedVersions`), I am more just using `openai` as an example here.

What should I specify as a `rangeStrategy`?

---

Here is my current `renovate.json5`:

```json5
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  extends: ["config:recommended"],
  rangeStrategy: "widen",
  lockFileMaintenance: {
    enabled: true,
  },
  "pre-commit": {
    enabled: true,
  },
  packageRules: [
    {
      matchUpdateTypes: ["lockFileMaintenance"],
      automerge: true,
    },
    {
      // group:allNonMajor, with automerge
      // We rely on Renovate's recommended behavior for major version updates
      groupName: "all non-major dependencies",
      groupSlug: "all-minor-patch",
      matchPackageNames: ["*"],
      matchUpdateTypes: ["minor", "patch"],
      automerge: true,
    },
  ],
}
```

**Comment by colton-freeman** [comment]
Not sure if this is exactly what you are looking for, but we have a rule in our packageManagers that looks like so 
```
    {
      "description": "Ignore any updates for docker image that starts with 37. ",
      "matchPackageNames": [
        "path/to/docker/registry"
      ],
      "allowedVersions": "!/^37\\./",
      "automerge": false
    },
```
This just keeps our podman image from bumping to anything that start with 37. since the registry we pull from messed up on one of their tags. This might be able to be expanded upon???
